lib/dao.js

```diff
@@ -1206,7 +1206,7 @@ DataAccessObject._coerce = function (where) {
           operator = 'regexp';
         } else if (operator === 'regexp' && val instanceof RegExp) {
           // Do not coerce regex literals/objects
-        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof RegExp)) {
+        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof Object)) {
```

Comment:
I think we should add another || for your case. Why is the RegExp check being changed?